src/lib.rs

```diff
@@ -1404,44 +1404,74 @@ fn get_the_effective_directive_for_inline_checks(type_: InlineCheckType) -> &'st
 /// <https://www.w3.org/TR/CSP/#script-pre-request>
 fn script_directives_prerequest_check(request: &Request, directive: &Directive) -> CheckResult {
     use CheckResult::*;
+    // Step 1. If request’s destination is script-like:
     if request_is_script_like(request) {
         let source_list = SourceList(&directive.value[..]);
+        // Step 1.1. If the result of executing § 6.7.2.3 Does nonce match source list? on
+        // request’s cryptographic nonce metadata and this directive’s value is "Matches", return "Allowed".
         if source_list.does_nonce_match_source_list(&request.nonce) == Matches {
             return Allowed;
         }
-        if source_list.does_integrity_metadata_match_source_list(&request.integrity_metadata) == Matches {
-            return Allowed;
+        // Step 1.2. If the result of executing § 6.7.2.4 Does integrity metadata match source list? on
+        // request’s integrity metadata and this directive’s value is "Does Not Match", return "Blocked".
+        if source_list.does_integrity_metadata_match_source_list(&request.integrity_metadata) == DoesNotMatch {
```

Comment:
Do not merge this PR yet, as it currently assumes the spec would be updated according to https://github.com/w3c/webappsec-csp/issues/728